tests/sagemaker/mock/__init__.py

```diff
@@ -0,0 +1,629 @@
+        """
+        endpoint_summaries = self.sagemaker_backend.list_endpoints()
+        return json.dumps({
+            'Endpoints': [summary.response_object for summary in endpoint_summaries]
+        })
+
+    def list_endpoint_configs(self):
+        """
+        Handler for the SageMaker "ListEndpointConfigs" API call documented here:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/API_ListEndpointConfigs.html.
+
+        This function does not support pagination. All endpoint configs are returned in a
+        single response.
+        """
+        # Note:
+        endpoint_config_summaries = self.sagemaker_backend.list_endpoint_configs()
+        return json.dumps({
+            'EndpointConfigs': [summary.response_object for summary in endpoint_config_summaries]
+        })
+
+    def list_models(self):
+        """
+        Handler for the SageMaker "ListModels" API call documented here:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/API_ListModels.html.
+
+        This function does not support pagination. All endpoint configs are returned in a
+        single response.
+        """
+        model_summaries = self.sagemaker_backend.list_models()
+        return json.dumps({
+            'Models': [summary.response_object for summary in model_summaries]
+        })
+
+    def create_model(self):
+        """
+        Handler for the SageMaker "CreateModel" API call documented here:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateModel.html.
+        """
+        model_name = self.request_params["ModelName"].encode("utf-8")
+        primary_container = self.request_params["PrimaryContainer"]
+        execution_role_arn = self.request_params["ExecutionRoleArn"]
+        tags = self.request_params.get("Tags", [])
+        vpc_config = self.request_params.get("VpcConfig", None)
+        new_model = self.sagemaker_backend.create_model(
+                model_name=model_name, primary_container=primary_container,
+                execution_role_arn=execution_role_arn, tags=tags, vpc_config=vpc_config,
+                region_name=self.region)
+        return json.dumps({
+            'ModelArn': new_model.arn
+        })
+
+    def describe_model(self):
+        """
+        Handler for the SageMaker "DescribeModel" API call documented here:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/API_DescribeModel.html.
+        """
+        model_name = self.request_params["ModelName"]
+        model_description = self.sagemaker_backend.describe_model(model_name)
+        return json.dumps(model_description.response_object)
+
+    def delete_model(self):
+        """
+        Handler for the SageMaker "DeleteModel" API call documented here:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/API_DeleteModel.html.
+        """
+        model_name = self.request_params["ModelName"]
+        self.sagemaker_backend.delete_model(model_name)
+        return ""
+
+
+class SageMakerBackend(BaseBackend):
```

Comment:
Out of curiosity, do you know how much work it would've been to implement these in moto?

Reply:
I'm not sure I fully understand the question. The implementation seems pretty similar to the Moto ECR library: https://github.com/spulec/moto/blob/master/moto/ecr/models.py